hyprpm/src/progress/CProgressBar.cpp

```diff
@@ -1,53 +1,42 @@
 #include "CProgressBar.hpp"
 
-#include <iostream>
 #include <algorithm>
 #include <cmath>
 #include <format>
 
-#include <sys/ioctl.h>
+#include <print>
 #include <stdio.h>
 #include <unistd.h>
 
 #include "../helpers/Colors.hpp"
 
-void CProgressBar::printMessageAbove(const std::string& msg) {
-    struct winsize w;
-    ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
-
-    std::string spaces;
-    for (size_t i = 0; i < w.ws_col; ++i) {
-        spaces += ' ';
-    }
+CProgressBar::CProgressBar() {
+    ioctl(STDOUT_FILENO, TIOCGWINSZ, &m_winsize);
+}
 
-    std::cout << "\r" << spaces << "\r" << msg << "\n";
+void CProgressBar::printMessageAbove(const std::string& msg) {
+    const std::string spaces(' ', m_winsize.ws_col);
+    std::println("\r{}\r{}", spaces, msg);
     print();
 }
 
 void CProgressBar::print() {
-    struct winsize w;
```

Comment:
why did you make this a member? what if it changes?

Reply:
You right, reverted